ff/serving/metadata/metadata.go

```diff
@@ -770,7 +788,49 @@ type MetadataServer struct {
 func NewMetadataServer(logger *zap.SugaredLogger) (*MetadataServer, error) {
 	logger.Debug("Creating new metadata server")
 	return &MetadataServer{
-		lookup: make(localResourceLookup),
+		lookup: TypesenseWrapper{
+			ResourceLookup: make(localResourceLookup),
+		},
+		Logger: logger,
+	}, nil
+}
+
+func NewMetadataTypesenseServer(logger *zap.SugaredLogger, port string, host string, apiKey string) (*MetadataServer, error) {
+	logger.Debug("Creating new metadata server")
+	client := typesense.NewClient(
+		typesense.WithServer("http://"+host+":"+port),
+		typesense.WithAPIKey(apiKey))
+	schema := &api.CollectionSchema{
+		Name: "feature",
```

Comment:
Will other resource types require different naming? If this is generic for all resource types (I assume it is) just call this "resource"